pkg/kubernetes/node/v1alpha1/node_test.go

```diff
@@ -0,0 +1,179 @@
+				t.Fatalf("Test %v failed: expected %v got %v", name, mock.expectedNodeLen, len(b.list.items))
+			}
+		})
+	}
+}
+
+func TestListBuilderFuncWithEmptyNodeList(t *testing.T) {
+	tests := map[string]struct {
+		nodeCount, expectedNodeLen int
+	}{
+		"Two nodes": {5, 5},
+	}
+	for name, mock := range tests {
+		name := name // pin it
+		mock := mock // pin it
+		t.Run(name, func(t *testing.T) {
+			b := ListBuilderFunc().WithAPIList(fakeEmptyNodeList(mock.nodeCount))
+			if mock.expectedNodeLen != len(b.list.items) {
+				t.Fatalf("Test %v failed: expected %v got %v", name, mock.expectedNodeLen, len(b.list.items))
+			}
+		})
+	}
+}
+
+func TestListBuilderWithAPIObjects(t *testing.T) {
+	tests := map[string]struct {
+		availableNodes  []string
+		expectedNodeLen int
+	}{
+		"Node set 2":  {[]string{"node1"}, 1},
+		"Node set 3":  {[]string{"node1", "node2"}, 2},
+		"Node set 4":  {[]string{"node1", "node2", "node3"}, 3},
+		"Node set 5":  {[]string{"node1", "node2", "node3", "node4"}, 4},
+		"Node set 6":  {[]string{"node1", "node2", "node3", "node4", "node5"}, 5},
+		"Node set 7":  {[]string{"node1", "node2", "node3", "node4", "node5", "node6"}, 6},
+		"Node set 8":  {[]string{"node1", "node2", "node3", "node4", "node5", "node6", "node7"}, 7},
+		"Node set 9":  {[]string{"node1", "node2", "node3", "node4", "node5", "node6", "node7", "node8"}, 8},
+		"Node set 10": {[]string{"node1", "node2", "node3", "node4", "node5", "node6", "node7", "node8", "node9"}, 9},
+	}
+	for name, mock := range tests {
+		name := name // pin it
+		mock := mock // pin it
+		t.Run(name, func(t *testing.T) {
+			b := ListBuilderFunc().WithAPIObject(fakeAPINodeList(mock.availableNodes).Items...)
+			if mock.expectedNodeLen != len(b.list.items) {
+				t.Fatalf("Test %v failed: expected %v got %v", name, mock.expectedNodeLen, len(b.list.items))
+			}
+		})
+	}
+}
+
+func TestListBuilderToAPIList(t *testing.T) {
+	tests := map[string]struct {
+		availableNodes  []string
+		expectedNodeLen int
+	}{
+		"Node set 1":  {[]string{}, 0},
+		"Node set 2":  {[]string{"node1"}, 1},
+		"Node set 3":  {[]string{"node1", "node2"}, 2},
+		"Node set 4":  {[]string{"node1", "node2", "node3"}, 3},
+		"Node set 5":  {[]string{"node1", "node2", "node3", "node4"}, 4},
+		"Node set 6":  {[]string{"node1", "node2", "node3", "node4", "node5"}, 5},
+		"Node set 7":  {[]string{"node1", "node2", "node3", "node4", "node5", "node6"}, 6},
+		"Node set 8":  {[]string{"node1", "node2", "node3", "node4", "node5", "node6", "node7"}, 7},
+		"Node set 9":  {[]string{"node1", "node2", "node3", "node4", "node5", "node6", "node7", "node8"}, 8},
+		"Node set 10": {[]string{"node1", "node2", "node3", "node4", "node5", "node6", "node7", "node8", "node9"}, 9},
+	}
+	for name, mock := range tests {
+		name := name // pin it
+		mock := mock // pin it
+		t.Run(name, func(t *testing.T) {
+			b := ListBuilderFunc().WithAPIList(fakeAPINodeList(mock.availableNodes)).List().ToAPIList()
```

Comment:
What is test function is testing.
If It is Only `ToAPIList()` then build builder struct using builder functions is not a good option.
This looks it dependent on `ListBuilderFunc()` ,` WithAPIList(...)`, `fakeAPINodeList()`,`List()` also
we can do something like this -
```go
nl := &NodeList{items: mock.availableNodes}
b := nl.ToAPIList()
```

Reply:
@imazik function name is representing `TestListBuilderToAPIList` so we have testing in way `ListBuilderFunc().WithAPIList(fakeAPINodeList(mock.availableNodes)).List().ToAPIList()`